tests/test_api.py

```diff
@@ -68,12 +68,13 @@ async def test_http_methods(client):
 @pytest.mark.parametrize(
     "url,pattern",
     [
-        ("https://foo.bar", "https://foo.bar"),
+        ("https://foo.bar", "https://foo.bar/"),
```

Comment:
I think we should support empty path like it was before here. Feels wrong to not match and require the pattern to have a trailing slash if not the httpx client needs it.